WakaTime.py

```diff
@@ -279,21 +327,41 @@ def enough_time_passed(now, is_write):
             or (now - _last_hb['timestamp'] > (2 if is_write else HEARTBEAT_FREQUENCY * 60)))
 
 
+@lru_cache(maxsize=4)
+def guessProjectName(filename):
```

Comment:
To solve the problem of cached settings
```suggestion
def guessProjectName(
    filename,
    truncate_trail,
    use_project_folder,
    project_prefix,
    project_postfix
):
```
A bit of a mouthful, but serves the purpose of cache invalidation. Still better then performing the whole calculation each time, since _most of the time_ user won't be changing settings, but would just work on a single file for hours.